x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Chunk.java

```diff
@@ -29,75 +31,55 @@
 
-    public static final int DEFAULT_NUM_CHUNKS = Integer.MAX_VALUE;
-    public static final int DEFAULT_CHUNK_SIZE = 300;
-
-    private final Expression field, options;
-
-    static final String NUM_CHUNKS = "num_chunks";
-    static final String CHUNK_SIZE = "chunk_size";
-
-    public static final Map<String, DataType> ALLOWED_OPTIONS = Map.of(NUM_CHUNKS, DataType.INTEGER, CHUNK_SIZE, DataType.INTEGER);
-
-    @FunctionInfo(returnType = "keyword", preview = true, description = """
-        Use `CHUNK` to split a text field into smaller chunks.""", detailedDescription = """
-            Chunk can be used on fields from the text famiy like <<text, text>> and <<semantic-text, semantic_text>>.
-            Chunk will split a text field into smaller chunks, using a sentence-based chunking strategy.
-            The number of chunks returned, and the length of the sentences used to create the chunks can be specified.
-        """, examples = { @Example(file = "chunk", tag = "chunk-with-field", applies_to = "stack: preview 9.3.0") })
+    static final int DEFAULT_CHUNK_SIZE = 300;
+    public static final ChunkingSettings DEFAULT_CHUNKING_SETTINGS = new SentenceBoundaryChunkingSettings(DEFAULT_CHUNK_SIZE, 0);
+
+    private final Expression field, chunkingSettings;
+
+    @FunctionInfo(
+        returnType = "keyword",
+        preview = true,
+        description = """
+            Use `CHUNK` to split a text field into smaller chunks.""",
+        detailedDescription = """
+                Chunk can be used on fields from the text famiy like <<text, text>> and <<semantic-text, semantic_text>>.
+                Chunk will split a text field into smaller chunks, using a sentence-based chunking strategy.
+                The number of chunks returned, and the length of the sentences used to create the chunks can be specified.
+            """,
+        examples = {
+            @Example(file = "chunk", tag = "chunk-with-field", applies_to = "stack: preview 9.3.0"),
+            @Example(file = "chunk", tag = "chunk-with-chunking-settings", applies_to = "stack: preview 9.3.0") }
+    )
     public Chunk(
         Source source,
         @Param(name = "field", type = { "keyword", "text" }, description = "The input to chunk.") Expression field,
         @MapParam(
-            name = "options",
-            params = {
-                @MapParam.MapParamEntry(
-                    name = "num_chunks",
-                    type = "integer",
-                    description = "The number of chunks to return. Defaults to return all chunks."
-                ),
-                @MapParam.MapParamEntry(
-                    name = "chunk_size",
-                    type = "integer",
-                    description = "The size of sentence-based chunks to use. Defaults to " + DEFAULT_CHUNK_SIZE
-                ), },
-            description = "Options to customize chunking behavior.",
+            name = "chunking_settings",
```

Comment:
if we like to refer to these as `chunking_settings` when we talk about them, that is fine.
but from an ES|QL POV these are called options in every other function.
let's reuse the same pattern and call these options.

Reply:
Discussed offline, we will use the name `chunking_settings` but ensure better documentation.